### PR TITLE
Centipede integration setup

### PR DESCRIPTION
**Pull request by Alan32Liu**
WIP: Project setups and tests.

The test cases are a bit confusing, particularly about the usage of [buckets `a-b`](https://github.com/google/clusterfuzz/blob/c5d3476ef597addbfc8848b2c0ae1be1e428a55c/src/clusterfuzz/_internal/tests/appengine/handlers/cron/project_setup_test.py#L1684), `c-d`, etc.
Is there any hint about how they are used? Thanks!
@jonathanmetzman @oliverchang

### CODE REVIEW

**Comment by oliverchang** [comment]
> buckets `a-b`

Ignore these ones. The only test you need to update for now are the ones here:

https://github.com/google/clusterfuzz/blob/c5d3476ef597addbfc8848b2c0ae1be1e428a55c/src/clusterfuzz/_internal/tests/appengine/handlers/cron/project_setup_test.py#L116

**Comment by Alan32Liu** [comment]
There are two unit test errors that I am not sure how to fix:
```

File: src/clusterfuzz/_internal/tests/appengine/handlers/cron/project_setup_test.py [hunk 1]
```diff
@@ -314,6 +319,13 @@ def test_execute(self):
             'sanitizers': ['none'],
             'architectures': ['i386', 'x86_64'],
         }),
+        ('lib9', {
+            'homepage': 'http://example.com',
+            'primary_contact': 'primary@example.com',
+            'main_repo': 'https://github.com/google/main-repo',
+            'fuzzing_engines': ['centipede',],
+            'sanitizers': ['addres',],
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/tests/appengine/handlers/cron/project_setup_test.py:327]: address is not spelled correctly.

File: src/clusterfuzz/_internal/bot/tasks/fuzz_task.py [hunk 2]
```diff
@@ -852,8 +852,8 @@ def store_fuzzer_run_results(testcase_file_paths, fuzzer, fuzzer_command,
     fuzzer_return_code_string = 'Fuzzer timed out.'
   truncated_fuzzer_output = truncate_fuzzer_output(fuzzer_output,
                                                    data_types.ENTITY_SIZE_LIMIT)
-  console_output = u'%s: %s\n%s\n%s' % (bot_name, fuzzer_return_code_string,
-                                        fuzzer_command, truncated_fuzzer_output)
+  console_output = '%s: %s\n%s\n%s' % (bot_name, fuzzer_return_code_string,
```
Comment (jonathanmetzman) [inline src/clusterfuzz/_internal/bot/tasks/fuzz_task.py:855]: Maybe just change this to an f-string.

File: src/clusterfuzz/_internal/system/environment.py [hunk 3]
```diff
@@ -671,6 +671,11 @@ def is_kernel_fuzzer_job(job_name=None):
   return get_engine_for_job(job_name) == 'syzkaller'
 
 
+def is_centipede_fuzzer_job(job_name=None):
+  """Return true if the current job uses Centipede."""
```
Comment (jonathanmetzman) [inline src/clusterfuzz/_internal/system/environment.py:675]: "True"

File: src/local/butler/scripts/setup.py [hunk 4]
```diff
@@ -84,6 +84,18 @@
...
+CENTIPEDE_TEMPLATE = """MAX_FUZZ_THREADS = 1
+MAX_TESTCASES = 2
+FUZZ_TEST_TIMEOUT = 8400
+TEST_TIMEOUT = 65
+WARMUP_TIMEOUT = 65
+BAD_BUILD_CHECK = False
+THREAD_ALIVE_CHECK_INTERVAL = 1
+REPORT_OOMS_AND_HANGS = True
+ENABLE_GESTURES = False
+THREAD_DELAY = 30.0
```
Comment (jonathanmetzman) [inline src/local/butler/scripts/setup.py:96]: @oliverchang do we know if this stuff is necessary? Seems weird (but understandable given CF's history) that we don't default to no for this stuff that is chrome only.

**Review by jonathanmetzman — APPROVED** [review]
lgtm

File: src/local/butler/scripts/setup.py (hunk 4 above)
Comment (Alan32Liu) [reply src/local/butler/scripts/setup.py:96]: Thanks for mentioning this!
Happy to remove this part if it is unnecessary, otherwise all addressed : )

Comment (oliverchang) [reply src/local/butler/scripts/setup.py:96]: I think a lot of these are indeed necessary. Many of the defaults are still from early Chrome days.

**Review by oliverchang — APPROVED** [review]

**Pull request merged**